File: abstract-factory/php/MySQLDatabaseFactory.php
<?php

namespace AbstractFactory;

use AbstractFactory\IDatabaseFactory;
use AbstractFactory\Connection;
use AbstractFactory\Command;
use AbstractFactory\MySQLConnection;
use AbstractFactory\MySQLCommand;

class MySQLDatabaseFactory implements IDatabaseFactory
{
    function createConnection(): Connection
    {
        return new MySQLConnection();
    }

    function createCommand(): Command
    {
        return new MySQLCommand();
    }
}

File: abstract-factory/dart/mysql_database_factory.dart
import 'command.dart';
import 'connection.dart';
import 'i_database_factory.dart';
import 'mysql_command.dart';
import 'mysql_connection.dart';

final class MysqlDatabaseFactory implements IDatabaseFactory {
  const MysqlDatabaseFactory();

  @override
  Command createCommand() {
    return const MysqlCommand();
  }

  @override
  Connection createConnection() {
    return const MysqlConnection();
  }
}
